plugin-server/src/types.ts

```diff
@@ -72,6 +72,34 @@ export enum KafkaSaslMechanism {
     ScramSha512 = 'scram-sha-512',
 }
 
+export enum PluginServerMode {
+    ingestion = 'ingestion',
+    ingestion_overflow = 'ingestion-overflow',
```

Comment:
To be consistent with variable casing we could use camel case instead of snake case here.